src/data-structures/Bytes.huff

```diff
@@ -0,0 +1,114 @@
+#define macro CONCAT_MEMORY(mem_ptr1, mem_ptr2) = takes(2) returns(1) {
+  // assert that mem_ptr2 >= mem_ptr1 + 0x20
+
+  // TODO: keep free_loc_pos at the end of the stack to return it
+  msize msize                       // [free_loc_pos, free_loc_pos]
+  <mem_ptr1> mload <mem_ptr2> mload // [len2, len1, free_loc_posi, free_loc_pos]
+  add swap1 mstore                  // [free_loc_pos]
+  msize                             // [index(j), free_loc_pos]
+  <mem_ptr1> dup1 0x20 add          // [index(i), mem_ptr1, index(j), free_loc_pos]
+  swap1 mload                       // [len1, index(i), index(j), free_loc_pos]
```

Comment:
Isn't it easier to say that `mload`ing the mem_ptr1 puts `len1` on the stack ? Then in the context of the looping, it becomes `len_left` because it's agnostic from pointer 1 or 2

Reply:
makes sense, resolving